controller/resource/csp_billing_adapter.go

```diff
@@ -42,9 +42,9 @@ func GetCspConfig() api.RESTFedCspSupportResp {
 	now := time.Now()
 	if obj, err = global.ORCH.GetResource(RscTypeConfigMap, NvAdmSvcNamespace, "csp-config"); err == nil {
 		if cm, ok := obj.(*corev1.ConfigMap); cm == nil || !ok {
-			err = fmt.Errorf("Error: Unknown type")
+			log.WithFields(log.Fields{"err": "Error: Unknown type"}).Error()
```

Comment:
in original code err is set to resp.NvError & resp is returned to caller.
this change makes caller think there is no NvError in GetCspConfig()

Reply:
a simplified code as below..
The number-1 and number-2 err assignment has no effect, no code have chance to use the value.
Because the err will be assigned in number-4 and it will use be used in number-5 for the final return value.

```
func GetCspConfig() api.RESTFedCspSupportResp {
	var err error

	if obj, err = global.ORCH.GetResource(RscTypeConfigMap, NvAdmSvcNamespace, "csp-config"); err == nil {
		if cm, ok := obj.(*corev1.ConfigMap); cm == nil || !ok {
			err = fmt.Errorf("Error: Unknown type")   1️⃣
		} else {
			err = fmt.Errorf("No billing data")       2️⃣
			if cm.Data != nil {
				if value, ok := cm.Data["data"]; ok {
					resp.CspConfigData = value
					var cspConfig tCspConfig
					if err = json.Unmarshal([]byte(value), &cspConfig); err == nil {   3️⃣
						if json_data, e := json.Marshal(cspConfig.Versions); e == nil {
							resp.AdapterVersions = string(json_data)
						}
						resp.CspErrors = cspConfig.Errors
						resp.CspConfigFrom = "local cluster"
						if tExpire, err = time.Parse("2006-01-02T15:04:05.000000-07:00", cspConfig.Expire); err == nil {

						}
					} else {
						log.WithFields(log.Fields{"err": err}).Error()
					}
				}
			}
		}
	}

	if obj, err = global.ORCH.GetResource(RscTypeConfigMap, NvAdmSvcNamespace, "metering-archive"); err == nil {  4️⃣
		if cm, ok := obj.(*corev1.ConfigMap); cm == nil || !ok {
			err = fmt.Errorf("Error: Unknown type")
		} else if cm.Data != nil {
			resp.MeteringArchiveData, _ = cm.Data["archive"]
		}
	}


	if err != nil || len(resp.CspErrors) > 0 {     5️⃣
		if err != nil { 
			resp.NvError = err.Error()
		}
		log.WithFields(log.Fields{"compliant": resp.Compliant, "nvError": resp.NvError, "expire": dataExpiredErr, "cspErrors": resp.CspErrors}).Error()
	}

	return resp
}

```

Reply:
similar situation as previous one.